include/circt/Dialect/FIRRTL/FIRRTLUtils.h

```diff
@@ -165,6 +165,17 @@ inline FIRRTLType mapBaseType(FIRRTLType type,
       });
 }
 
+/// Return a FIRRTLType of appropriate "const"-ness, based on its existing
+/// const status and the provided update function.
+inline FIRRTLType mapConstType(FIRRTLType type,
```

Comment:
It seems like this function is used the exact same way at all callsites (`return isConst || parentAggregateType.isConst()`). 

Maybe what would be better is more specific helper functions on the aggregate types, something like `getElementTypePreservingConst`.

I don't think I've run into a situation where I've wanted something more general than that yet.

Reply:
You're right, thank you!  mapBaseType is useful for working with ref-types, but yeah at best this really only helps by dealing with RefType where `getConstType` is not supported.  I'll see about simplifying or just dropping!

Reply:
Oh, looking more closely I see what you mean, yeah that would be a nice feature "propagate your const to your element", 100%!

Will give that a quick shot, since adding new aggregates anyway.  Thanks!

Reply:
Done, thanks!